### PR TITLE
feat: added validation for the Advanced Settings page

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**

### CODE REVIEW

**Review by ruzniaievdm — APPROVED** [review]

File: src/advanced-settings/AdvancedSettings.jsx [hunk 1]
```diff
@@ -35,19 +54,30 @@ const AdvancedSettings = ({ intl, courseId }) => {
...
     setEditedSettings({});
     showSaveSettingsPrompt(false);
   };
 
   const handleUpdateAdvancedSettingsData = () => {
-    dispatch(updateCourseAppSetting(courseId, parseArrayOrObjectValues(editedSettings)));
-    showSaveSettingsPrompt(false);
-    window.scrollTo({ top: 0, behavior: 'smooth' });
+    validateAdvancedSettingsData(editedSettings, setErrorFields)
+        .then(() => {
```
Comment (viktorrusakov) [inline src/advanced-settings/AdvancedSettings.jsx:69]: nit: something wrong with indents? should be 2 spaces I think

File: src/advanced-settings/AdvancedSettings.jsx [hunk 2]
```diff
@@ -5,28 +5,47 @@ import { Container, Button, Layout } from '@edx/paragon';
...
   const savingStatus = useSelector(getSavingStatus);
   const proctoringExamErrors = useSelector(getProctoringExamErrors);
+  const settingsWithSendErrors = useSelector(getSendRequestErrors) || {};
   const dispatch = useDispatch();
   const [saveSettingsPrompt, showSaveSettingsPrompt] = useState(false);
   const [showDeprecated, setShowDeprecated] = useState(false);
+  const [errorModal, showErrorModal] = useState(false);
   const [editedSettings, setEditedSettings] = useState({});
+  const [errorFields, setErrorFields] = useState([]);
+  const [successAlert, showSuccessAlert] = useState(false);
```
Comment (viktorrusakov) [inline src/advanced-settings/AdvancedSettings.jsx:31]: boolean states should not be named like this, please rename to `[showSuccessAlert, setShowSuccessAlert]`. same with other states added in this PR

File: src/advanced-settings/modal-error/ModalError.test.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import ModalError from './ModalError';
+
+describe('ModalError', () => {
+    const defaultProps = {
```
Comment (viktorrusakov) [inline src/advanced-settings/modal-error/ModalError.test.jsx:7]: nit: indent should be 2 spaces?

File: src/advanced-settings/modal-error/ModalErrorListItem.test.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import ModalErrorListItem from './ModalErrorListItem';
+
+describe('ModalErrorListItem', () => {
+    const settingsData = {
```
Comment (viktorrusakov) [inline src/advanced-settings/modal-error/ModalErrorListItem.test.jsx:6]: nit: indent should be 2 spaces?

File: src/utils.js [hunk 5]
```diff
@@ -84,6 +96,46 @@ export function useAppSetting(settingName) {
   return [settingValue, saveSetting];
 }
 
+export const validateAdvancedSettingsData = (object, setErrorFields) => {
```
Comment (viktorrusakov) [inline src/utils.js:99]: this function should not be located in global utils, please move it to `advanced-settings` folder

File: src/advanced-settings/AdvancedSettings.jsx [hunk 6]
```diff
@@ -5,28 +5,47 @@ import { Container, Button, Layout } from '@edx/paragon';
...
 
   useEffect(() => {
     dispatch(fetchCourseAppSettings(courseId));
     dispatch(fetchProctoringExamErrors(courseId));
-  }, [courseId]);
+
+    switch (savingStatus) {
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/advanced-settings/AdvancedSettings.jsx:37]: Thanks for the proposals 👍 
About proposal №3. If we have a lot of "Bad requests". We need to show these errors in the error modal window.

File: src/advanced-settings/AdvancedSettings.jsx (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply src/advanced-settings/AdvancedSettings.jsx:69]: indent: 'off' :)

**Pull request merged**